packages/@expo/cli/src/run/android/resolveLaunchProps.ts

```diff
@@ -25,16 +25,22 @@ async function getMainActivityAsync(projectRoot: string): Promise<string> {
   return activity.$['android:name'];
 }
 
-export async function resolveLaunchPropsAsync(projectRoot: string): Promise<LaunchProps> {
+export async function resolveLaunchPropsAsync(
+  projectRoot: string,
+  options: { appIdSuffix?: string }
+): Promise<LaunchProps> {
   // Often this is ".MainActivity"
   const mainActivity = await getMainActivityAsync(projectRoot);
 
   const packageName = await new AndroidAppIdResolver(projectRoot).getAppIdFromNativeAsync();
+  const packageNameWithSuffix = options.appIdSuffix
+    ? `${packageName}.${options.appIdSuffix}`
+    : packageName;
   const launchActivity = `${packageName}/${mainActivity}`;
```

Comment:
Shouldn't `launchActivity` also be updated? I think in most cases it'll be `${packageNameWithSuffix}/${packageName}${mainActivity}`, e.g. cli from react-native-community prepares `activityToLaunch` in [such way](https://github.com/react-native-community/cli/blob/a414d98683f059162e9e106faab39f40ccf5228c/packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L27) and later uses [${applicationIdWithSuffix}\/${activityToLaunch}](https://github.com/react-native-community/cli/blob/a414d98683f059162e9e106faab39f40ccf5228c/packages/cli-platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L42) in `adbArgs`.

Moreover, `launchActivity` shouldn't be passed to [openCustomRuntimeAsync](https://github.com/expo/expo/blob/07d561b42396fd95452417899e775b993a384181/packages/%40expo/cli/src/run/android/runAndroidAsync.ts#L68)?
```ts
await manager.getDefaultDevServer().openCustomRuntimeAsync(
  'emulator',
  {
    applicationId: props.packageName,
    launchActivity: props.launchActivity
  },
  { device: props.device.device }
);
```

Reply:
Yes, you are right, I missed the case where app is opened from appId. I have borrowed the code from react-native about `launchActivity` and added `launchActivity` to `openCustomRuntimeAsync`.